### PR TITLE
reafactor: separate and test GradebookFilters from Gradebook component

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -**
Adds unit tests/snapshots for Assignment filter components, which involved a bit of splitting them up to make them more testable.
Separates out StudentGroupsFilters and CourseGradesFilters into their own components, and adds associated unit tests.

**What changed?**
Jest, lint, and webpack config updates to allow module path imports (no more '../../../../')
Jest renderer updates
new GradebookFilters sub-view folder/component w/ AssignmentFilter, AssignmentTypeFilter, AssignmentGradeFilter, CourseGradeFilter, and StudentGroupsFilter components split out into it.
Shared PercentGroup and SelectGroup help enforce more modern Paragon form usage.

Snapshot and unit tests for all files in the filters folder.

JIRA: 
[EDUCATOR-5665](https://openedx.atlassian.net/browse/EDUCATOR-5665)
[EDUCATOR-5666](https://openedx.atlassian.net/browse/EDUCATOR-5666)

**Developer Checklist**
- [x] Test suites passing
- [ ] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**

[ How should a reviewer test this PR? ]

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: src/components/Gradebook/GradebookFilters/index.jsx
Comment (jansenk) [inline src/components/Gradebook/GradebookFilters/index.jsx:26]: I don't totally follow. Are we keeping this in both this component's state and redux state? why?

File: src/components/Gradebook/GradebookFilters/index.jsx
Comment (jansenk) [inline src/components/Gradebook/GradebookFilters/index.jsx:45]: What's this?

File: src/components/Gradebook/GradebookFilters/test.jsx
Comment (jansenk) [inline src/components/Gradebook/GradebookFilters/test.jsx:46]: "calls props.updateQueryParams"

File: src/components/Gradebook/GradebookFilters/test.jsx
Comment (jansenk) [inline src/components/Gradebook/GradebookFilters/test.jsx:55]: tbh, I don't totally understand the value of having only these "basic snapshots". What is this test asserting? I mean, I get that it's asserting that the snapshot hasn't changed, but I guess I don't see what the value of that is. Wouldn't we want, like, a snapshot of all of the different possible states of this component?

File: src/components/Gradebook/GradebookFilters/test.jsx
Comment (jansenk) [inline src/components/Gradebook/GradebookFilters/test.jsx:51]: Should we test that the state has been updated?

File: src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/index.jsx
Comment (jansenk) [inline src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/index.jsx:41]: I know this is a refactor but this seems like it's a bug .... should the Assignment Types dropdown really be disabled when there are no selectable assignments for that Assignment Type? That doesn't make sense to me.

File: src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/index.jsx
Comment (jansenk) [inline src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/index.jsx:70]: Since we're only using this as a boolean, why not make this something like `assignmentsOfSelectedTypeExist: selectableAssignmentLabels(state).length === 0`?

I think you can also use `selectableAssignments(state)` rather than `selectableAssignmentLabels(state)` since we don't care about the additional mapping the latter does.

File: src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/test.jsx
Comment (jansenk) [inline src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/test.jsx:22]: are these random caps a thing you do on purpose? i notice them in other tests too.

File: src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/test.jsx
Comment (jansenk) [inline src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/test.jsx:63]: 'handleChange'

also, why again do you do this?

File: src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/test.jsx
Comment (jansenk) [inline src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/test.jsx:118]: where do this and state.filters.track come from? it looks like above you only set assignmentType. could they be passing because they're both `undefined`?

**Review by jansenk — COMMENTED** [review]
I'm only about halfway through reviewing everything but I wanted to submit my comments so fr before I sign off for the night

File: src/components/Gradebook/GradebookFilters/index.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/index.jsx:26]: to make react controlled component happy.  Doesn't like being controlled by external value when it also allows toggling of that value.

File: src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/index.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/index.jsx:41]: There are... a few logic inconsistencies with the current implementation that I feel should be reviewed. (some "disabled" filters can also still be saved in the filterBadges...

File: src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/index.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/index.jsx:70]: Fair.  This is just flat copying over the existing mapping.
Though, I will say if we were doing the first, I want it in a selector.  I hate the extent to which we are directly querying the redux store shape. Just not worth fixing as part of this ticket.

File: src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/test.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/test.jsx:22]: Yes.  Ensures we aren't getting algorithmig/auto text.  Just makes the tests a little bit more secure.

File: src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/test.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/test.jsx:63]: ah, I update string. 
I do this so that they show up in the snapshot.

File: src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/test.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/test.jsx:118]: sigh, yes.  I fix

File: src/components/Gradebook/GradebookFilters/index.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/index.jsx:45]: react components must return a *single* jsx object.  In order to support this without adding an outer div (which can affect display) you can add a react Fragment, which is shorthanded to <> </>

File: src/components/Gradebook/GradebookFilters/test.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/test.jsx:46]: danke.

File: src/components/Gradebook/GradebookFilters/test.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/test.jsx:51]: yes.  That feature was added to fix a bug yesterday and I did't add test.  adding now, thx.

File: src/components/Gradebook/GradebookFilters/test.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/test.jsx:55]: We do want a snapshot of all different states that render into snapshots.  In this case, there is no conditional rendering, so there really is only one state.  For non-conditional render components, a single "basic" snapshot with nominal props is generally sufficient.

Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/test.jsx:55]: still verifies basic layout, structure, and connection of methods (if mocked appropriately)

File: src/components/Gradebook/index.jsx
Comment (muselesscreator) [reply src/components/Gradebook/index.jsx:434]: ah, should be.  Was for testing.

Comment (muselesscreator) [reply src/components/Gradebook/index.jsx:434]: removing here and for tracks below

File: src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/test.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/AssignmentTypeFilter/test.jsx:118]: Actually, these 2 tests shouldn't even be in here.  sigh

File: src/components/Gradebook/GradebookFilters/StudentGroupsFilter/test.jsx
Comment (jansenk) [inline src/components/Gradebook/GradebookFilters/StudentGroupsFilter/test.jsx:21]: could you change these ids to be like, some other numbers, so it's clearer when you're accessing them by index

File: src/components/Gradebook/GradebookFilters/StudentGroupsFilter/test.jsx
Comment (jansenk) [inline src/components/Gradebook/GradebookFilters/StudentGroupsFilter/test.jsx:82]: "if no cohort is found"

File: src/components/Gradebook/GradebookFilters/StudentGroupsFilter/__snapshots__/test.jsx.snap
Comment (jansenk) [inline src/components/Gradebook/GradebookFilters/StudentGroupsFilter/__snapshots__/test.jsx.snap:1]: Should we have a snapshot where Cohorts selector is disabled if there are no cohorts?

File: src/components/Gradebook/GradebookFilters/index.jsx
Comment (nsprenkle) [inline src/components/Gradebook/GradebookFilters/index.jsx:47]: Is there a reason you chose this over just wrapping with the `<Collapsible>` tags? It just feels like an extra level of redirection.

File: webpack.dev.config.js
Comment (nsprenkle) [inline webpack.dev.config.js:2]: Similar note here to the jest config, I would defer to what is already in `master` since I fixed some eslint issues and standardized the import orders between these 2 files.

File: webpack.prod.config.js
Comment (nsprenkle) [inline webpack.prod.config.js:8]: Same note as `webpack.dev.config.js` and `jest.config.js`

**Review by nsprenkle — APPROVED** [review]
Looks good. Minor stylistic nits, also I would propose using existing webpack and jest configs for standardization & eslint reasons.

File: src/components/Gradebook/GradebookFilters/index.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/index.jsx:47]: just less repeated css-work.

File: src/components/Gradebook/GradebookFilters/AssignmentFilter/test.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/AssignmentFilter/test.jsx:63]: I find that spacing can get a bit wonky with grouping/nesting

File: src/components/Gradebook/GradebookFilters/AssignmentGradeFilter/test.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/AssignmentGradeFilter/test.jsx:43]: fair.  It used to.

File: src/components/Gradebook/GradebookFilters/CourseGradeFilter/test.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/CourseGradeFilter/test.jsx:14]: just making the paragon components show up more simplified in the snapshot

File: src/components/Gradebook/GradebookFilters/StudentGroupsFilter/index.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/StudentGroupsFilter/index.jsx:110]: yep

File: src/components/Gradebook/GradebookFilters/StudentGroupsFilter/test.jsx
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/StudentGroupsFilter/test.jsx:21]: yep

File: src/components/Gradebook/GradebookFilters/StudentGroupsFilter/__snapshots__/test.jsx.snap
Comment (muselesscreator) [reply src/components/Gradebook/GradebookFilters/StudentGroupsFilter/__snapshots__/test.jsx.snap:1]: yeah... was holding back because that should just be coming from a selector-based prop, but it makes sense to go ahead and add the test while that conditional is in there.

**Pull request merged**
